messageBuilder.js

```diff
@@ -0,0 +1,65 @@
+const messages = {
+  prTitle: 'PR Title',
+  prBody: 'PR Description',
+  spaceBetweenHashtagAndDigit: 'PR contains a space between the hashtag and digit.',
+  noIssueReference: 'Issue Reference (`#<issue-number>`) missing.',
+  missingGithubKeyword: 'GitHub Keyword missing: Refer [here](https://help.github.com/articles/closing-issues-via-commit-messages/#keywords-for-closing-issues) for a list of accepted keywords.',
+};
+
+function isKeywordCheckerEnabled() {
+  return process.env.ENABLE_KEYWORD_CHECKER !== undefined &&
+    process.env.ENABLE_KEYWORD_CHECKER.toLowerCase() === 'true';
+}
+
+/*
+* Formats message as a GFMD level two unordered list item
+*/
+function getFormattedMessageLevelTwoUnordered(message) {
+  return `   * ${message}\n`;
+}
+
+/*
+* Formats message as a GFMD level two unordered list item
+*/
+function getFormattedMessageLevelOneOrdered(message) {
+  return `1. ${message}\n`;
+}
+
+function buildTitleFeedback(violations) {
+  let message = '';
+  if (violations === undefined) return message;
+  if (violations.main === true) {
+    message += getFormattedMessageLevelOneOrdered(messages.prTitle);
+  }
+  return message;
+}
+
+function buildDescriptionFeedback(violations) {
+  let message = '';
+  if (violations === undefined) return message;
+  if (violations.main === true) {
+    message += getFormattedMessageLevelOneOrdered(messages.prBody);
+    if (isKeywordCheckerEnabled()) {
+      Object.keys(violations.details).forEach((key) => {
+        message += getFormattedMessageLevelTwoUnordered(messages[key]);
+      });
+    }
+  }
+  return message;
+}
+
+module.exports = {
+  getFormattedMessageLevelOneOrdered,
+  getFormattedMessageLevelTwoUnordered,
+  messages,
+  /*
+  * Returns a string containing the formatted feedback message
+  */
+  getFeedbackMessage(username, violations) {
+    const feedback = `Hi @${username}, these parts of your pull request do not appear to follow our [contributing guidelines](${process.env.CONTRIBUTING_GUIDELINES}):\n\n`;
+
+    return feedback +
+      buildTitleFeedback(violations.title) +
```

Comment:
Hang on, if it defaults to the empty object then i would be checking for the empty object instead of undefined. In that case, undefined would be an actual error